receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,300 @@
+	// For example, if tag_name is not specified and the key is git_sha,
+	// then the attribute name will be `k8s.pod.annotations.git_sha`.
+	// When key_regex is present, tag_name supports back reference to both named capturing and positioned capturing.
+	// For example, if your pod spec contains the following labels,
+	//
+	// app.kubernetes.io/component: mysql
+	// app.kubernetes.io/version: 5.7.21
+	//
+	// and you'd like to add tags for all labels with prefix app.kubernetes.io/ and also trim the prefix,
+	// then you can specify the following extraction rules:
+	//
+	// extract:
+	//   labels:
+	//     - tag_name: $$1
+	//       key_regex: kubernetes.io/(.*)
+	//
+	// this will add the `component` and `version` tags to the spans or metrics.
+	TagName string `mapstructure:"tag_name"`
+
+	// Key represents the key (annotation, label or etc.) name. It uses exact match.
+	Key string `mapstructure:"key"`
+	// KeyRegex is a regular expression used to extract a Key that matches the regex.
+	// Out of Key or KeyRegex, only one option is expected to be configured at a time.
+	KeyRegex string `mapstructure:"key_regex"`
+
+	// Regex is an optional field used to extract a sub-string from a complex field value.
+	// The supplied regular expression must contain one named parameter with the string "value"
+	// as the name. For example, if your pod spec contains the following annotation,
+	//
+	// kubernetes.io/change-cause: 2019-08-28T18:34:33Z APP_NAME=my-app GIT_SHA=58a1e39 CI_BUILD=4120
+	//
+	// and you'd like to extract the GIT_SHA and the CI_BUILD values as tags, then you must
+	// specify the following two extraction rules:
+	//
+	// extract:
+	//   annotations:
+	//     - tag_name: git.sha
+	//       key: kubernetes.io/change-cause
+	//       regex: GIT_SHA=(?P<value>\w+)
+	//     - tag_name: ci.build
+	//       key: kubernetes.io/change-cause
+	//       regex: JENKINS=(?P<value>[\w]+)
+	//
+	// this will add the `git.sha` and `ci.build` resource attributes.
+	Regex string `mapstructure:"regex"`
+}
+
+func (c Config) Validate() error {
+	err := c.Discovery.Validate()
+	if c.Discovery.Mode != ModeDaemonSetStdout &&
+		len(c.Include) == 0 {
+		err = multierr.Append(err, fmt.Errorf("include must be specified when discovery mode is not daemonset_stdout"))
+	}
+	if c.StartAt != StartAtBeginning && c.StartAt != StartAtEnd {
+		err = multierr.Append(err, fmt.Errorf("invalid start_at: %s", c.StartAt))
+	}
+	return err
+}
+
+// K8sSourceConfig allows specifying how to discover containers to collect logs from.
+type K8sSourceConfig struct {
+	// Mode represents the mode of the k8slog receiver.
+	// Valid values are:
+	// - "daemonset-stdout": (default) otel is deployed as a daemonset and collects logs from stdout of containers.
+	// - "daemonset-file": otel is deployed as a daemonset and collects logs from files inside containers.
+	// - "sidecar": otel is deployed as a sidecar and collects logs from files.
+	Mode string `mapstructure:"mode"`
```

Comment:
In the same idea of limiting the configuration options to start, would it make sense to limit the modes to start as well? I think starting with only `daemonset-stdout` would still make this a very useful component.

Reply:
I'll do it soon